crates/trident/src/stream.rs

```diff
@@ -0,0 +1,220 @@
+                        None => 0,
+                    };
+                    let high = match args.get("high") {
+                        Some(v) => v.as_u64().unwrap_or(u64::MAX),
+                        None => u64::MAX,
+                    };
+
+                    let filtered: Vec<tera::Value> = a
+                        .into_iter()
+                        .filter(|item| {
+                            if let Some(size) = item.get("size").and_then(|s| s.as_u64()) {
+                                size >= low && size <= high
+                            } else {
+                                false
+                            }
+                        })
+                        .collect();
+
+                    Ok(tera::Value::Array(filtered))
+                }
+                x => Ok(x),
+            }
+        }
+    }
+
+    let disks = tera::Value::Array(
+        lsblk::list()
+            .unwrap_or_default()
+            .into_iter()
+            .filter(|b| b.blkdev_type == BlockDeviceType::Disk)
+            .filter(|d| {
+                d.name.starts_with("sd") || d.name.starts_with("nvme") || d.name.starts_with("vd")
+            })
+            .map(|b| {
+                let mut m = serde_json::Map::new();
+                m.insert("name".into(), tera::Value::String(b.name.clone()));
+                m.insert(
+                    "path".into(),
+                    tera::Value::String(format!("/dev/{}", b.name)),
+                );
+                m.insert("size".into(), tera::Value::Number(b.size.into()));
+
+                println!(
+                    "Detected disk: name='{}' path='/dev/{}' size={}",
+                    b.name, b.name, b.size
+                );
+                tera::Value::Object(m)
+            })
+            .collect(),
+    );
+
+    let mut tera = Tera::default();
+    tera.register_filter("size_range", SizeRange);
+    tera.add_raw_template("config.yaml", template)?;
+
+    let mut context = tera::Context::new();
+    context.insert("disks", &disks);
+    context.insert("KB", &1024);
+    context.insert("MB", &(1024 * 1024));
+    context.insert("GB", &(1024 * 1024 * 1024));
+
+    Ok(tera.render("config.yaml", &context)?)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_expand_template_basic_context() {
+        assert_eq!(
+            expand_template(
+                r#"
+kb_value: {{ KB }}
+mb_value: {{ MB }}
+gb_value: {{ GB }}
+"#
+            )
+            .unwrap(),
+            r#"
+kb_value: 1024
+mb_value: 1048576
+gb_value: 1073741824
+"#
+        );
+    }
+
+    #[test]
+    fn test_expand_template_detect_disks_function() {
+        assert!(expand_template(
+            r#"
+disks: {{ disks }}
+"#
+        )
+        .is_ok());
+    }
+
+    #[test]
+    fn test_expand_template_size_range_filter() {
+        let result = expand_template(
+            r#"
+small_disks: {{ disks | size_range(high=1073741824) }}
```

Comment:
nit: this syntax may be more confusing, i'd suggest just having a function called `disk_selector` or similar which will read the disks internally, instead of using a value and a filter